common/djangoapps/student/views.py

```diff
@@ -1804,6 +1805,28 @@ def create_account_with_params(request, params):
     return new_user
 
 
+def _enroll_user_in_pending_courses(user_id):
+    """
+    Enroll student in any pending courses he/she may have.
+    """
+    student = User.objects.filter(id=user_id)
```

Comment:
@adampalay Yeah your right. Its an extra database call, I have removed it. The user is created in `_do_create_account` call. I have passed it directly into `_enroll_user_in_pending_courses`.